### DISCUSSION TITLE
Is there a way to get the mean annual maximum of monthly data with an xarray function?

### DISCUSSION THREAD

**Discussion by pecos27** [pydata/xarray#5854] [Q&A] [answered]
Is it possible in xarray to get the mean annual maximum of monthly data?
I am hoping to get the maximum temperature per year, so for each year there would be one maximum and then can take the annual mean to get the mean annual maximum.

I tried get the maxima per year via a loop, but I was wondering if there is another way do it just through xarray?

Example code below:

```
ds = xr.tutorial.open_dataset('rasm')

# getting 1980 - 1982 
Tair = ds.Tair[4:-8]

#Tair_max = Tair.max(dim='time')


# attempt using a loop

maximum_every_12_months = np.zeros((int(Tair.shape[0]/12),Tair.shape[1],Tair.shape[2]))

for i in range(12):
    print(i)
    maximum_every_12_months[i,:,:] = np.max(Tair[i::12,:,:],axis=0)

```

**Comment by TomNicholas** [comment, marked as answer]
Have you seen the [GroupBy functionality?](http://xarray.pydata.org/en/stable/user-guide/groupby.html) If you added a year coordinate to your data then you could do something like `da.groupby("year").max()`.

**Reply by spencerkclark** [reply]
Indeed `groupby` is the way to go.  Further, as long as the time coordinate is composed of some form of datetime object, you can do something like this:
```python
da.groupby("time.year").max()
```
In other words, there is no need to explicitly create your own `"year"` coordinate.

**Reply by pecos27** [reply]
Thank you! yes this works perfectly. much nicer than using a loop :)
